lib/hammer_cli_katello/content_view_purge.rb

```diff
@@ -84,7 +84,8 @@ def execute
     private def old_unused_versions
       return @old_unused_versions if @old_unused_versions
 
-      all_versions = index(:content_view_versions, :content_view_id => options['option_id'])
+      all_versions = index(:content_view_versions, :content_view_id => options['option_id'],
```

Comment:
My worry about using a page size of 1000 is that it might lead to some very large API calls that could potentially time out or cause other issues.  Also, if the user for some reason has 2000 versions, they'll hit the bug again.

I posted an idea I had below. It seems to work okay. I loop through the content view versions, ensuring that all pages get accounted for.  That way, we get the benefit of a safely-sized API call too.

```suggestion
      all_versions = []
      per_page = 100
      page = 1

      loop do
        versions = index(:content_view_versions, :content_view_id => options['option_id'], :per_page => per_page, :page => page)
        all_versions << versions
        break if versions.count < per_page
        page += 1
      end
      all_versions.flatten!
```

Reply:
This looks better than my approach, will go with this ty.